examples/platform/nxp/common/matter_button/source/ButtonApp.cpp

```diff
@@ -27,20 +27,17 @@ extern "C" {
 /**
  * @brief Flag to describe if the button handles are predefined in SDK.
  *
- * Set to true by default. Platforms that do not have this support should
- * disable the flag, which will enable handle definition in this file.
+ * By default assuming button handles are predefined in the SDK
+ * If not, flag CHIP_APP_BUTTON_HANDLE_SDK_NOT_PREDEFINED should be defined
  */
-#ifndef CONFIG_APP_BUTTON_HANDLE_SDK_PREDEFINED
-#define CONFIG_APP_BUTTON_HANDLE_SDK_PREDEFINED 1
-#endif
 
-#if !CONFIG_APP_BUTTON_HANDLE_SDK_PREDEFINED
+#ifdef CONFIG_CHIP_APP_BUTTON_HANDLE_SDK_NOT_PREDEFINED
```

Comment:
within the matter SDK, we prefer to not use `ifdef` or `if defined` because that forces correct includes to be added. Otherwise code silently compiles and maybe wrongly so because of missing includes

Reply:
Changes applied as suggested.